code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -1060,6 +1060,41 @@
 
 /////////////////////////////////////////////////////////////////////////////////////////////
 
+////////////////////
+/// Rally Hive
+///////////////////
+/datum/action/xeno_action/activable/rally_hive
+	name = "Rally Hive"
+	action_icon_state = "rally_hive"
+	mechanics_text = "Rallies the hive to a congregate at a target location, along with an arrow pointer. Gives the Hive your current health status. 60 second cooldown."
+	use_state_flags = XACT_TARGET_SELF
+	ability_name = "rally hive"
+	plasma_cost = 0
+	keybind_signal = COMSIG_XENOABILITY_RALLY_HIVE
+	cooldown_timer = 60 SECONDS
+
+///Halve the cooldown for Hiveminds as their relative omnipresence means they can actually make use of this lower cooldown.
+/datum/action/xeno_action/activable/rally_hive/New()
+	. = ..()
+	if(isxenohivemind(owner))
+		cooldown_timer = 30 SECONDS
+
+/datum/action/xeno_action/activable/rally_hive/use_ability(atom/A)
+
+	var/mob/living/carbon/xenomorph/X = owner
+
+	X.face_atom(A) //Face towards the target so we don't look silly
```

Comment:
No need to typecast

Reply:
Need it for the Hive reference for the xeno_message